stan/math/prim/prob/wiener5_lpdf.hpp

```
@@ -0,0 +1,839 @@
+  }
+}
+
+/**
+ * Calculate the 'result' term and its sign for a wiener5 density or gradient
+ *
+ * @tparam Density Whether the calculation is for the density
+ * @tparam GradW Whether the calculation is for gradient w.r.t. 'w'
+ * @tparam T_y type of scalar variable
+ * @tparam T_a type of boundary separation
+ * @tparam T_w type of relative starting point
+ * @tparam T_nsmall type of term number_small_terms
+ * @tparam T_nlarge type of term number_large_terms
+ *
+ * @param y A scalar variable; the reaction time in seconds
+ * @param a The boundary separation
+ * @param w_value The relative starting point
+ * @param n_terms_small_t The n_terms_small_t term
+ * @param n_terms_large_t The n_terms_large_t term
+ * @return 'result' sum and its sign
+ */
+template <bool Density, GradientCalc GradW, typename T_y, typename T_a,
+          typename T_w, typename T_nsmall, typename T_nlarge>
+inline auto wiener5_log_sum_exp(T_y&& y, T_a&& a, T_w&& w_value,
+                                T_nsmall&& n_terms_small_t,
+                                T_nlarge&& n_terms_large_t) noexcept {
+  const auto y_asq = y / square(a);
+  const auto w = 1.0 - w_value;
+  const bool small_n_terms_small_t
+      = Density ? (2 * n_terms_small_t <= n_terms_large_t)
+                : (2 * n_terms_small_t < n_terms_large_t);
+  const auto scaling = small_n_terms_small_t ? inv(2.0 * y_asq) : y_asq / 2.0;
+  using ret_t = return_type_t<T_y, T_a, T_w, T_nsmall, T_nlarge>;
+  ret_t fplus = NEGATIVE_INFTY;
+  ret_t fminus = NEGATIVE_INFTY;
+  int current_sign;
+  if (small_n_terms_small_t) {
+    constexpr double mult = Density ? 1.0 : 3.0;
+    if (GradW) {
+      const auto offset = y_asq;
+      for (auto k = n_terms_small_t; k >= 1; k--) {
+        const auto wp2k = w + 2.0 * k;
+        const auto wm2k = w - 2.0 * k;
+        const auto sqwp2k_mo = square(wp2k) - offset;
+        if (sqwp2k_mo > 0) {
+          const auto wp2k_quant = log(sqwp2k_mo) - square(wp2k) * scaling;
```

Should this throw an error? log(<= 0) is undefined I don't really understand why it would be correct to just drop it

---

I don't get your point. This should not throw an error. 
If (sqwp2k_mo > 0), we go with log(sqwp2k_mo)
and if it is <0, we go with log(-...),
to ensure that the argument of log is always >0.